### PR TITLE
MF-998 - Add Twins service to Makefile and docker-compose.yml

### PR DESCRIPTION
**Pull request by darkodraskovic**
Integrates the twins service in the overall mainflux docker composition system.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -190,3 +190,7 @@ MF_POSTGRES_READER_DB_SSL_MODE=disable
 MF_POSTGRES_READER_DB_SSL_CERT=""
 MF_POSTGRES_READER_DB_SSL_KEY=""
 MF_POSTGRES_READER_DB_SSL_ROOT_CERT=""
+
+### Twins
+MF_TWINS_HTTP_PORT=9021
+MF_TWINS_MQTT_URL=tcp://mqtt-adapter:${MF_MQTT_ADAPTER_PORT}
```
Comment (manuio) [inline .env]: There is no new line at the end of the file

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -260,3 +260,31 @@ services:
...
+      MF_TWINS_THING_ID: ${MF_TWINS_THING_ID}
+      MF_TWINS_THING_KEY: ${MF_TWINS_THING_KEY}
+      MF_TWINS_CHANNEL_ID: ${MF_TWINS_CHANNEL_ID}
+      MF_NATS_URL: ${MF_NATS_URL}
+      MF_TWINS_MQTT_URL: ${MF_TWINS_MQTT_URL}
+    ports:
+      - ${MF_TWINS_HTTP_PORT}:${MF_TWINS_HTTP_PORT}
+    networks:
+      - mainflux-base-net
+    env_file:
```
Comment (manuio) [inline docker/docker-compose.yml]: This is necessary?

**Review by manuio — CHANGES_REQUESTED** [review]

File: docker/docker-compose.yml (hunk 2 above)
Comment (nmarcetic) [reply docker/docker-compose.yml]: I would also avoid additional .env file. If you must put it in global .env twins section

**Review by nmarcetic — CHANGES_REQUESTED** [review]

File: docker/docker-compose.yml (hunk 2 above)
Comment (manuio) [reply docker/docker-compose.yml]: It's not an additional one, it's the same. My question is if it really works if you run the composition from a different location than where the .env is.

Comment (nmarcetic) [reply docker/docker-compose.yml]: @manuio Its called `.env-twins` its a new file. He used both .env and new .env-twins no ?

Comment (manuio) [reply docker/docker-compose.yml]: Right, I didn't see that he use both. In any case no need to specify the `.env` path

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -260,3 +260,31 @@ services:
       - ${MF_COAP_ADAPTER_PORT}:${MF_COAP_ADAPTER_PORT}/tcp
     networks:
       - mainflux-base-net
+
+  twins:
+    image: mainflux/twins:latest
+    container_name: mainflux-twins
+    depends_on:
+      - authn
+      - nats
```
Comment (bbokun) [inline docker/docker-compose.yml]: please be aware that the twins service depends on a mongo database, what means that either the dependency has to be to the standard mongodb from the writer, or - better, a separate twins-db has to be created and added to the docker-compose file

File: .env (hunk 1 above)
Comment (bbokun) [inline .env]: is the protocol "tcp" neccessary? it works also without it.

Comment (bbokun) [inline .env]: looking at other sections in the .env file, all the possible variables are listed
here we miss the following:
MF_TWINS_LOG_LEVEL=debug
MF_TWINS_SERVER_CERT=""
MF_TWINS_SERVER_KEY=""
MF_TWINS_DB_NAME=mainflux
MF_TWINS_DB_HOST=twins-db
MF_TWINS_DB_PORT=27017
MF_TWINS_SINGLE_USER_EMAIL=""
MF_TWINS_SINGLE_USER_TOKEN=""
MF_TWINS_CLIENT_TLS=""
MF_TWINS_CA_CERTS=""

**Review by bbokun — CHANGES_REQUESTED** [review]

**Pull request closed without merging**